R/db.R

```diff
@@ -219,3 +261,54 @@ setMethod("dbUnquoteIdentifier", c("Microsoft SQL Server", "SQL"),
     x <- gsub("(\\[)([^\\.]+?)(\\])", "\\2", x)
     callNextMethod( conn, x, ... )
   })
+
+#' SQL Server specific implementation.
+#'
+#' Local temp tables are stored as
+#' \code{ [tempdb].[dbo].[#name]________(padding using underscores)[numeric identifier] }
+#'
+#' True if:
+#' - If catalog_name is supplied it must equal "temdb" or "%" ( wildcard )
+#' - Name must start with "#" followd by a non-"#" character
+#' @rdname SQLServer
+#' @usage NULL
+setMethod("isTempTable", c("Microsoft SQL Server", "character"),
+  function(conn, name, catalog_name = NULL, schema_name = NULL, ...) {
+    if ( !is.null(catalog_name) &&
+        catalog_name != "%" &&
+        length(catalog_name ) > 0 &&
+        catalog_name != "tempdb" ) {
+      return(FALSE)
+    }
+
+    if ( !grepl("^[#][^#]", name ) ) {
+      return(FALSE)
+    }
+    return(TRUE)
+})
+
+#' SQL Server specific implementation.
+#'
+#' Will warn user if `temporary` is set to TRUE but table name does not conform
+#' to local temp table naming conventions.  If writing to a global temp table, user
+#' should not set the temporary flag to TRUE.
+#'
+#' In both cases a simple CREATE TABLE statement is used / the table identifier is
+#' is the differentiator ( viz-a-viz creating a non-temp table ).
+#' @rdname SQLServer
+#' @usage NULL
+setMethod("sqlCreateTable", "Microsoft SQL Server",
+  function(con, table, fields, row.names = NA, temporary = FALSE, ..., field.types = NULL) {
+    if ( temporary && !isTempTable( con, table ) )
+    {
+      warning(paste0("Temporary flag is set to true, but name argument seems",
+                     " to point to non-local-temp table"))
+    }
+    table <- dbQuoteIdentifier(con, table)
+    fields <- createFields(con, fields, field.types, row.names)
+
+    SQL(paste0(
+      "CREATE TABLE ", table, " (\n",
+      "  ", paste(fields, collapse = ",\n  "), "\n)\n"
+    ))
```

Comment:
Could you just use `callNextMethod()` here?

Reply:
Good idea; done.
Thank you